apis/v1beta1/validation/gateway.go

```diff
@@ -118,3 +119,17 @@ func ValidateTLSCertificateRefs(listeners []gatewayv1b1.Listener, path *field.Pa
 	}
 	return errs
 }
+
+// ValidateListenerNames validates the names of the listeners
+// must be unique within the Gateway
+func ValidateListenerNames(listeners []gatewayv1b1.Listener, path *field.Path) field.ErrorList {
```

Comment:
This func should be called by `validateGatewayListeners()` to be included in the admission webhook.

Reply:
Added in the `validateGatewayListeners` above